pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -83,50 +95,156 @@ func NewCronJobController(kubeClient clientset.Interface) (*CronJobController, e
 		sjControl:  &realSJControl{KubeClient: kubeClient},
 		podControl: &realPodControl{KubeClient: kubeClient},
 		recorder:   eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cronjob-controller"}),
-	}
-
+		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "cronjob"),
+	}
+	cjbInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    jm.addCronjob,
+		UpdateFunc: jm.updateCronjob,
+		DeleteFunc: jm.deleteCronjob,
+	})
+
+	jm.cjobLister = cjbInformer.Lister()
+	jm.jobLister = jobInformer.Lister()
+	jm.podLister = podInformer.Lister()
+	jm.cjobSynced = cjbInformer.Informer().HasSynced
+	jm.jobSynced = jobInformer.Informer().HasSynced
+	jm.podSynced = podInformer.Informer().HasSynced
+	jm.syncHandler = jm.syncCronJob
 	return jm, nil
 }
 
 // Run the main goroutine responsible for watching and syncing jobs.
 func (jm *CronJobController) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
+	defer jm.queue.ShutDown()
+
+	if !controller.WaitForCacheSync("cronjob", stopCh, jm.cjobSynced, jm.jobSynced, jm.podSynced) {
+		return
+	}
+
 	glog.Infof("Starting CronJob Manager")
-	// Check things every 10 second.
-	go wait.Until(jm.syncAll, 10*time.Second, stopCh)
+
+	// we need add all the CronJobs to the queue periodically to make sure we can start a job according to the schedule
+	go wait.Until(func() {
+		glog.Info("Start enqueue all CronJobs")
+		cjobs, err := jm.cjobLister.List(labels.Everything())
```

Comment:
That's the crucial part that needs proper handling. Re-queuing all cronjobs does not solve the problem here. That's not the approach I'm looking for.

Reply:
Could you describe again (because I do not remember it from the KubeCon 2018 - Copenhagen talk) what you would like to see as a proper solution? :)